File: sorting/selection_sort.py
def get_min_index(array, frm):
    min_index = frm
    for i in range(frm, len(array)):
        if array[i] < array[min_index]:
            min_index = i
    return min_index

def selection_sort(array):
    for i in range(len(array)):
        min_index = get_min_index(array, i)
        if array[i] > array[min_index]:
            temp = array[i]
            array[i] = array[min_index]
            array[min_index] = temp    
    return array


print(get_min_index([2, 1, 8, 7, 10], 0))
print(selection_sort([2, 1, 8, 7, 10]))

File: sorting/selectionSort.js
function getMinIndex(array, frm){
    let indx = frm;
    for (let i = frm+1; i < array.length; i++){
        if (array[i] < array[indx]){
            indx = i;
        }
    }
    return indx;
}

//console.log(getMinIndex([8, 3, 10, 7, 5, 20], 0));

function SelectionSort(array){
    for (let i = 0; i < array.length; i++){
        let minIndex = getMinIndex(array, i);
        if (array[i] > array[minIndex]){
            let temp = array[i];
            array[i] = array[minIndex];
            array[minIndex] = temp;
        }
    }
    return array;
}

console.log(SelectionSort([8, 3, 10, 7, 5, 20]));